rflx/model/message.py

```diff
@@ -1632,7 +1679,28 @@ def prune_dangling_fields(
                 break
 
             field, inner_message = inner_messages.pop(0)
-            inner_message = inner_message.prefixed(f"{field.name}_")
+            inner_message, message_attribute_locations = replace_message_attributes(
+                inner_message.prefixed(f"{field.name}_")
+            )
+
+            message_next_fields = message.outgoing(field)
+            if len(message_attribute_locations) > 0 and (
+                len(message_next_fields) > 1 or message_next_fields[0].target != FINAL
+            ):
+                self.error.append(
+                    "Message types with message attribute may only be used for fields preceeding"
+                    " FINAL",
```

Comment:
The meaning of "FINAL" is most likely not clear for a user. There is also a typo in "preceding".

Reply:
I changed that in 877fd78d to "...may only be used for last field". Not sure whether this is that much better as users could confuse it with the physically last field in the message (OTOH, if the underlying graph concept isn't understood, every more complex explanation will probably not help either).